idaes/core/surrogate/onnx_surrogate.py

```diff
@@ -0,0 +1,250 @@
+            output_labels=output_labels,
+            input_bounds=input_bounds,
+            input_scaler=input_scaler,
+            output_scaler=output_scaler,
+        )
+
+        self._onnx_model = onnx_model
+
+    class Formulation(Enum):
+        FULL_SPACE = 1
+        REDUCED_SPACE = 2
+        RELU_BIGM = 3
+        RELU_COMPLEMENTARITY = 4
+
+    def populate_block(self, block, additional_options=None):
+        """
+        Method to populate a Pyomo Block with the keras model constraints.
+
+        Args:
+           block: Pyomo Block component
+              The block to be populated with variables and/or constraints.
+           additional_options: dict or None
+              If not None, then should be a dict with the following keys;
+              'formulation': KerasSurrogate.Formulation
+              The formulation to use with OMLT. Possible values are FULL_SPACE,
+              REDUCED_SPACE, RELU_BIGM, or RELU_COMPLEMENTARITY (default is FULL_SPACE)
+        """
+        formulation = additional_options.pop(
+            "formulation", ONNXSurrogate.Formulation.REDUCED_SPACE
+        )
+        omlt_scaling, scaled_input_bounds = self.generate_omlt_scaling_objecets()
+
+        # omlt takes *scaled* input bounds as a dictionary with int keys
+        input_bounds = dict(enumerate(self.input_bounds().values()))
+        scaled_input_bounds = omlt_scaling.get_scaled_input_expressions(input_bounds)
+        scaled_input_bounds = {i: tuple(bnd) for i, bnd in scaled_input_bounds.items()}
+
+        # TODO: remove this once new OMLT 1.2 is made available and includes tanh support
+        # overrides default available activation functions for ONNX, tanh is not listed in 1.1 but is supported
+        omltio.onnx_parser._ACTIVATION_OP_TYPES = [
+            "Relu",
+            "Sigmoid",
+            "LogSoftmax",
+            "Tanh",
+        ]
+
+        net = load_onnx_neural_network(
+            self._onnx_model,
+            scaling_object=omlt_scaling,
+            input_bounds=scaled_input_bounds,
+        )
+
+        if formulation == ONNXSurrogate.Formulation.FULL_SPACE:
+            formulation_object = FullSpaceSmoothNNFormulation(net)
+        elif formulation == ONNXSurrogate.Formulation.REDUCED_SPACE:
+            formulation_object = ReducedSpaceSmoothNNFormulation(net)
+        elif formulation == ONNXSurrogate.Formulation.RELU_BIGM:
+            formulation_object = ReluBigMFormulation(net)
+        elif formulation == ONNXSurrogate.Formulation.RELU_COMPLEMENTARITY:
+            formulation_object = ReluComplementarityFormulation(net)
+        else:
+            raise ValueError(
+                'An unrecognized formulation "{}" was passed to '
+                "KerasSurrogate.populate_block. Please pass a valid "
+                "formulation.".format(formulation)
+            )
+        self.populate_block_with_net(block, formulation_object)
+
+    def evaluate_surrogate(self, inputs):
```

Comment:
Why is this present but only raises NotImplementError?

Reply:
Left it here for future implementation, but setting up ONNX model to solve outside idaes did not seem trivial at time of first commit. (e.g. there did not seem to be something as simple as _keras_model.predict.